src/test/java/it/gov/pagopa/payment/controller/BasePaymentControllerIntegrationTest.java

```diff
@@ -132,8 +141,23 @@ private TransactionResponse createTrxSuccess(TransactionCreationRequest trxReque
         return trxCreated;
     }
 
-    private void checkTransactionStored(TransactionResponse trxCreated) {
+    private void checkTransactionStored(TransactionResponse trxCreated) throws Exception {
         TransactionInProgress stored = checkIfStored(trxCreated.getId());
+        SyncTrxStatusDTO syncTrxStatusResult =extractResponse(getStatusTransaction(trxCreated.getId(), trxCreated.getMerchantId(), trxCreated.getAcquirerId()),HttpStatus.OK, SyncTrxStatusDTO.class);
+        assertAll(() -> {
+                    Assertions.assertEquals(trxCreated.getId(), syncTrxStatusResult.getId());
+                    Assertions.assertEquals(trxCreated.getTrxCode(), syncTrxStatusResult.getTrxCode());
+                    Assertions.assertEquals(trxCreated.getInitiativeId(), syncTrxStatusResult.getInitiativeId());
+                    Assertions.assertEquals(trxCreated.getMerchantId(), syncTrxStatusResult.getMerchantId());
+                    Assertions.assertEquals(trxCreated.getIdTrxIssuer(), syncTrxStatusResult.getIdTrxIssuer());
+                    Assertions.assertEquals(trxCreated.getTrxDate(), syncTrxStatusResult.getTrxDate());
+                    Assertions.assertEquals(trxCreated.getAmountCents(), syncTrxStatusResult.getAmountCents());
+                    Assertions.assertEquals(trxCreated.getAmountCurrency(), syncTrxStatusResult.getAmountCurrency());
+                    Assertions.assertEquals(trxCreated.getMcc(), syncTrxStatusResult.getMcc());
+                    Assertions.assertEquals(trxCreated.getAcquirerId(), syncTrxStatusResult.getAcquirerId());
+                    Assertions.assertEquals(trxCreated.getStatus(), syncTrxStatusResult.getStatus());
+                });
```

Comment:
as done in row 262, use the mapper to assert the equals instead of repeating all the equality assertions (as alternative, you could re-use the assertions written in the mapper tests)!

Otherwise you should update these assertions each time you add a field to the mapped DTO!
And this is not the main purpose of this test class!